src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJobs.java

```diff
@@ -0,0 +1,156 @@
+package org.kiwiproject.dropwizard.util.job;
+
+import static com.google.common.collect.Maps.newConcurrentMap;
+import static java.util.Objects.isNull;
+import static org.kiwiproject.base.KiwiPreconditions.checkArgument;
+import static org.kiwiproject.base.KiwiPreconditions.checkArgumentNotNull;
+import static org.kiwiproject.base.KiwiStrings.f;
+
+import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.dropwizard.setup.Environment;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.dropwizard.util.config.JobSchedule;
+import org.kiwiproject.dropwizard.util.health.MonitoredJobHealthCheck;
+
+/**
+ * A set of utilities to assist in setting up MonitoredJobs with health checks.
+ */
+@Slf4j
+@UtilityClass
+public class MonitoredJobs {
+
+    @VisibleForTesting
+    static final Map<String, JobContext> JOBS = newConcurrentMap();
```

Comment:
Just use `new ConcurrentHashMap<>()` - there's no real advantage to use Guava for this (and this is exactly and all that `newConcurrentMap()` does)